resource/src/main/java/com/epam/learn/resource/service/StorageService.java

```diff
@@ -0,0 +1,42 @@
+package com.epam.learn.resource.service;
+
+import com.epam.learn.resource.domain.Storage;
+import com.epam.learn.resource.domain.StorageType;
+import io.github.resilience4j.circuitbreaker.annotation.CircuitBreaker;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.NoSuchElementException;
+
+@Slf4j
+@Service
+public class StorageService {
+
+    private final StorageServiceClient serviceClient;
+
+    public StorageService(StorageServiceClient serviceClient) {
+        this.serviceClient = serviceClient;
+    }
+
+    @CircuitBreaker(name = "storageBreaker", fallbackMethod = "getStorageStub")
```

Comment:
I would name it `getStorageByTypeStub`.

Reply:
You can define circuit breaker as part of client itself.

Reply:
renamed, left circuit breaker as it is for now (can change in the next commit, if you think its mandatory)